File: 879-maximize-distance-to-closest-person/maximize-distance-to-closest-person.cpp
class Solution {
public:
    int maxDistToClosest(vector<int>& seats) {
        int l = 0, d = 0;
        for (auto r = 0; r < seats.size(); r++) {
            if (!seats[l] && seats[r]) {
                d = max(d, r - l);
                l = r;
            }
            else if (seats[l] && seats[r]) {
                d = max(d, (r - l) / 2);
                l = r;
            }
        }
        if (l + 1 != seats.size())
            d = max(d, (int)(seats.size() - l - 1));
        return d;
    }
};

File: 879-maximize-distance-to-closest-person/maximize-distance-to-closest-person.go
func maxDistToClosest(seats []int) int {
    s := 0
    maxD := 0

    for f := 1; f < len(seats); f++ {
        if seats[s] == 0 && seats[f] == 1 ||
            seats[s] == 1 && seats[f] == 0 && f == len(seats) - 1 {
                maxD = max(maxD, f - s)
                s = f
        } else if seats[s] == 1 && seats[f] == 1 {
            maxD = max(maxD, (f - s) / 2)
            s = f
        }
    }

    return maxD
}